homeassistant/components/media_player/sonos.py

```diff
@@ -25,23 +26,19 @@
 import homeassistant.helpers.config_validation as cv
 from homeassistant.util.dt import utcnow
 
-REQUIREMENTS = ['SoCo==0.13']
+REQUIREMENTS = [
+    'https://github.com/SoCo/SoCo/archive/'
+    '9f848e7a2c73aebb0f6d0b09008b17990f4ffabc.zip#SoCo==0.14a5']
```

Comment:
A needed third party fix is on track to be released in SoCo 0.14 on February 17th. So feel free to review now, even though the `REQUIREMENTS` are not completely ready yet.

Reply:
That is okay, we had this before also some times :)

Reply:
I'm told it is no longer allowed to introduce Github requirements so I will hold off merging this PR until SoCo 0.14 ships.

Reply:
It's okay. That is only for packages they want produce a packages. But in this case we need this version and it will go into mainstream later

Reply:
Well, that is a policy that makes sense to me but I think some consensus is needed here. This PR has been lingering for two months because I did ask about just this policy in Discord and IIRC both @balloob  and @armills said that my only options were to a) fork SoCo with my single patch added, or b) wait it out for a new SoCo release.

Reply:
I think we're all still trying to figure out what the best policy is. We're trying to push to get all of our dependencies into pypi, but maybe we over-corrected. I think where we're landing is that cases like this are probably OK, as long as there's a plan to eventually switch back to a normal release.

My only request would be to put a comment above `REQUIREMENTS` explaining that we can switch back to pypi once version 0.14 is released.

Reply:
If there a release planned for Feb 17th I think we should wait for that.

Reply:
My view is that requirements pointing to the upstream master branch could be allowed, at least in some cases. If you are unable to get your commit onto the upstream master branch, maybe you have a bad idea or maybe the project does indeed need a fork. When the patch _is_ accepted, though, I think it is impolite to fork a project just because one does not like the release schedule.

IMHO a personal fork is fairly likely to be abandoned once the PR is merged into HA. This creates extra work (for someone else) to get HA back on the upstream track and it also pollutes PyPI.

(above, "fork" means a separate PyPI project and not just a Github fork which is obviously never impolite)

@andrey-git That's a bit silly. The only reason that a SoCo release is imminent is that I have been maintaining this patch out-of-tree for several months, waiting for this moment to arrive. I fully expected to wait two more weeks but the question is how to handle such a situation the next time.

Reply:
We had the same situation on last update of this platform and we had do it like now. First we point to commit and after the release was out, we update to new release. I see no problem, sorry.

Reply:
One technical question I have is how to format the string so the package is not downloaded on each restart. The SoCo Github currently has [version "0.12+"](https://github.com/SoCo/SoCo/blob/9f848e7a2c73aebb0f6d0b09008b17990f4ffabc/soco/__init__.py#L22) and we are using "0.13" in current HA code.

Reply:
You need look how pip does install it: `pip list --format=freeze`

Reply:
Thanks, that helped me figure it out. But that apparently points to a disadvantage of a Github requirement: two different commits can have the same version and `pip` will not know to update.

Reply:
Yes